plumbing/object/commit_walker.go

```diff
@@ -197,26 +197,38 @@ func NewCommitAllIter(repoStorer storage.Storer, commitIterFunc func(*Commit) Co
 	commitsPath := list.New()
 	commitsLookup := make(map[plumbing.Hash]*list.Element)
 	head, err := storer.ResolveReference(repoStorer, plumbing.HEAD)
-	if err != nil {
+	if err != nil && err != plumbing.ErrReferenceNotFound {
 		return nil, err
+	} else if err != plumbing.ErrReferenceNotFound {
```

Comment:
@kuba-- maybe:
```suggestion
	} else if err == nil {
```

Is more clear?